viewer.py

```diff
@@ -654,29 +767,50 @@ def __exit__(self, *args):
         self.signal.connect(self.slot)
 
 
-
 if __name__ == '__main__':
 
     app = pg.mkQApp()
 
     v = AtlasViewer()
     v.show()
 
-    path = os.path.dirname(__file__)
+    path = os.path.dirname(os.path.realpath(__file__))
     atlasFile = os.path.join(path, "ccf.ma")
     labelFile = os.path.join(path, "ccf_label.ma")
 
     if os.path.isfile(atlasFile):
         atlas = metaarray.MetaArray(file=atlasFile, readAllData=True)
     else:
-        atlas = readNRRDAtlas()
-        writeFile(atlas, atlasFile)
+        try:
+            atlas = readNRRDAtlas()
+            writeFile(atlas, atlasFile)
+        except:
+            try:
+                print "Unexpected error when creating ccf.ma file with " + atlasFile
+                print traceback.print_exc()
```

Comment:
Aha! Got it. Here's a more straightforward arrangement:

        try:
            atlas = readNRRDAtlas()
            writeFile(atlas, atlasFile)
        except:
            print "Unexpected error when creating ccf.ma file with " + atlasFile
            if os.path.isfile(atlasFile):
                try:
                    print "Removing ccf.ma"
                    os.remove(atlasFile)
                except:
                    print "Error removing ccf.ma:"
                    sys.excepthook(*sys.exc_info())
            raise